File: tracerPython/src/core/evaluator.py
import types

from .util import FrameUtil, ExceptionUtil


class Evaluator:
    """
    Evaluates an expression in a frame scope.
    """

    @classmethod
    def evaluate(cls, frame: types.FrameType, expression: str):
        """
        Evaluates expressions against the frame scope, process possible exceptions if any is thrown.
        The expression is able to mutate the script state.
        """
        try:
            product = eval(expression, FrameUtil.globals(frame), FrameUtil.locals(frame))
        except Exception as e:
            product = ExceptionUtil.dump(e)

        return {
            'product': product
        }

File: tracerJava/src/core/Evaluator.java
package core;

import com.sun.jdi.event.LocatableEvent;
import core.util.ExceptionUtil;

/**
 * Evaluates an expression in an event scope.
 */
public final class Evaluator {

    private Evaluator() {
    }

    /**
     * Evaluates expressions against the frame scope, process possible exceptions if any is thrown.
     * The expression is able to mutate the script state.
     */
    public static Object evaluate(LocatableEvent event, String expression) {
        try {
            throw new UnsupportedOperationException("evaluation not implemented");
        } catch (Exception e) {
            return ExceptionUtil.dump(e);
        }
    }
}
